x/poolmanager/keeper.go

```diff
@@ -101,17 +101,18 @@ func (k Keeper) InitGenesis(ctx sdk.Context, genState *types.GenesisState) {
 
 	// We track taker fees generated in the module's KVStore.
 	// If the values were exported, we set them here.
-	// If the values were not exported, we initialize the tracker to zero and set the accounting height to the current height.
-	if !genState.TakerFeesTracker.TakerFeesToStakers.Empty() {
-		k.SetTakerFeeTrackerForStakers(ctx, genState.TakerFeesTracker.TakerFeesToStakers)
-	} else {
-		k.SetTakerFeeTrackerForStakers(ctx, sdk.NewCoins())
+	// If the values were not exported, we set the accounting height to the current height.
```

Comment:
Shouldn't we instead be expecting them to be set? Why would it not be exported? I think this might lead to some state-inconsistencies down the road

Reply:
Yeah agreed, changed to expect the height being set here https://github.com/osmosis-labs/osmosis/pull/7240/commits/69c2df12d485df96116a513fb8a7b9ea9c634f92